scripts/cluster_buildings.py

```diff
@@ -114,7 +114,9 @@ def get_central_points_geojson_with_buildings(
         ]
         centroids_building = np.array(centroids_building)
         # Apply KMeans clustering to group the buildings
-        kmeans = KMeans(n_clusters=n_clusters, random_state=0).fit(centroids_building)
+        kmeans = KMeans(n_clusters=n_clusters[grid_name], random_state=0).fit(
+            centroids_building
+        )
```

Comment:
Given the above, there would be the need to distinguish the case to allow the model to work with n_clusters=10 and n_clusters=dict(microgrid_1=3, microgrid=13).
If you need support let me know and can go in more details :)